### ISSUE TITLE
OSX, can't bundle into .app file, 'NoneType' object has no attribute 'rfind'

### ISSUE THREAD

**Issue by nekopep** [pyinstaller/pyinstaller#1880]
Hello,

I can't manage to bundle into .app under OSX with pyinstaller, even the simplest one.
If I had '--windowed' option, pyinstaller crash.

Please see log file below:
[build.txt](https://github.com/pyinstaller/pyinstaller/files/170510/build.txt)

The build script:
#/bin/sh
rm -rf dist/*
pyinstaller -D --windowed --strip  --log-level DEBUG test.py

The test.py file:
print "OK"

The test.spec file:
# -*- mode: python -*-

block_cipher = None

a = Analysis(['test.py'],
             pathex=['/Users/ldufrechou/Documents/test'],
             binaries=None,
             datas=None,
             hiddenimports=[],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          exclude_binaries=True,
          name='test',
          debug=False,
          strip=True,
          upx=True,
          console=False )
coll = COLLECT(exe,
               a.binaries,
               a.zipfiles,
               a.datas,
               strip=True,
               upx=True,
               name='test')
app = BUNDLE(coll,
             name='test.app',
             icon=None,
             bundle_identifier=None)

I'm using python:
Python 2.7.11 (default, Mar  1 2016, 18:40:10) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.

From macports and pyinstaller 3.1.1 from pip (with macports python as default python).

**Comment by nekopep** [comment]
At least I can help a little here:
utils.py line 193, dist_nm is None -> so crash the app.
            if is_darwin:
                dylib.mac_set_relative_dylib_deps(cachedfile, dist_nm)
I've a debugger so can try several thing, just I need help on how is used dist_nm.

Thx!

**Comment by nekopep** [comment]
(previous comment was on pyinstaller-develop.zip)

**Comment by nekopep** [comment]
Also removing UPX from my system make .app working again.
So related to upx handling under OSX

**Comment by tallforasmurf** [comment]
The build.txt file shown ends with "'NoneType' object has no attribute 'rfind'" which is the same error as reported in #1839 and #1877. If @nekopep  is correct and it is related to UPX, that might also apply to those other two issues?

**Comment by htgoebel** [comment]
@tallforasmurf No, these issues are not related to UPX. #1839 is a missing shared python library and in #1877 the logs says: "UPX is not available". Anyway thanks for linking these issues, so I did not need the search them :-)

**Comment by nekopep** [comment]
I've just work a little more on this issue, so to be exact the issue arise if strip=True and UPX is present on the system.
To workarond this I corrected line 221 in utils.py

if upx:
        if strip:
            fnm = checkCache(fnm, strip=True, upx=False, dist_nm=dist_nm)
in fact dist_nm was not set leading to crash.
With this patch it looks like upx is launched.
Still under OSX it does not look very effective... I got a lot of:
23251 INFO: Executing - upx --best -q /Users/xxxx/Library/Application Support/pyinstaller/bincache01_py27_64bit/_ssl.so
upx: /Users/xxxx/Library/Application Support/pyinstaller/bincache01_py27_64bit/_ssl.so: UnknownExecutableFormatException

**Comment by htgoebel** [comment]
@nekopep Is this still an issue with the current release? If it is solved, please close the ticket.
